### DISCUSSION TITLE
Login/Registration is not working

### DISCUSSION THREAD

**Discussion by Hutch79** [JackBailey/Lynx#64] [Q&A] [answered]
Hey all 👋 
I have a problem while login/registering or the first time.

I visited the page and tried to register a new account, but it loaded forever until I reloaded the page.
After that, I tried it again, but I got the message that new registrations are disabled (Intentional).
So I thought the registration was successful and tried to log in, unfortunately it again loads for ages, and therefore I can't use anything.

A restart of my Docker stack was unsuccessful as well, and there are no errors visible in the logs.

**Comment by JackBailey** [comment]
Hi, I think there might've been an issue between registering the account internally and returning back the response.
Please could you try running the following command to delete the account in Lynx and try again with the browser console open?

```console
docker exec -it DOCKER_NAME mongosh shortener --username DB_USER --password "DB_PASSWORD" --authenticationDatabase admin --eval "db.accounts.remove({'username': 'USERNAME'})"
```

Replacing:
-  DOCKER_NAME with your docker service name, probably lynx_db
- DB_USER with your MONGO_INITDB_ROOT_USERNAME setting
- DB_PASSWORD with your MONGO_INITDB_ROOT_PASSWORD setting
- USERNAME with the username of the account you tried to create, if you don't know it run the same command with --eval "db.accounts.find()" instead
 
 You shouldn't need to restart anything after running this command.

**Reply by Hutch79** [reply]
<img width="955" alt="image" src="https://github.com/Lynx-Shortener/Lynx/assets/42042811/c6b03a53-b3c7-4591-9ae0-f3021a892bfe">
Seams to be an issue with Umami.
I'll remove it quickly and try again.

**Reply by Hutch79** [reply]
It worked fine without umami.

**Reply by JackBailey** [reply]
Okay, that's interesting. I'll see if I can add some better error handling there then. Did you get any errors loading the page with umami enabled? What were your umami variables set to?

**Reply by Hutch79** [reply]
They were set to the URL and the id which I got from the umami script.

- UMAMI_SITEID=a649df21-xxxx-xxxx-xxxx-5923fc5eb8f2
- UMAMI_URL=https://xxx.example.ch/main.js

The screenshot sent above was with umami, after I tried to register myself.

**Reply by JackBailey** [reply]
Ohh okay I see the issue, [it's /script.js, not /main.js](https://docs.getlynx.dev/installation/environment-variables#umami)

**Reply by Hutch79** [reply]
It is possible to set another name in umami.
I also use it on WordPress and Bookstack sites and it works fine.
Also, umami got the visits from lynx, even tough lynx didn't seam to like it ^^'
![grafik](https://github.com/Lynx-Shortener/Lynx/assets/42042811/77d3e053-f5cd-4cde-8be4-649636e1a5be)

**Reply by JackBailey** [reply]
Oh right that's interesting, I don't see that option anywhere (on my selfhosted site atleast). Are you using Umami V2?

window.umami.track being undefined makes me think umami events aren't quite loading properly, what do you get if you just run `window.umami` in the browser console?

**Reply by Hutch79** [reply]
No, I still was o 1.40, but I just updated it ^^'
I'll test it again afterward.

You can change the name with an env var `TRACKER_SCRIPT_NAME: main`

**Reply by Hutch79** [reply, marked as answer]
I just updated to umami v2, and now it works without any errors in the browser console.

**Reply by Hutch79** [reply]
Is it intended that it does only send the Lynx pages to Umami and no data for the redirect links?

**Reply by JackBailey** [reply]
Great, okay. Glad it's working for you. I'll add a warning to the console for trying to use the events with Umami V1.

**Reply by JackBailey** [reply]
> Is it intended that it does only send the Lynx pages to Umami and no data for the redirect links?

Yeah, as the redirects are usually handled by the backend, they're not able to be tracked. Proper analytics for Lynx including redirects are planned, just won't be added too soon.

**Reply by Hutch79** [reply]
Hmm, ok.
The analytics would be the only thing missing for me atm ^^
Do you plan to create your own statistics, or will it be in umami as well?

I don't understand a lot about Webdev, but I think umami could be easier since you wouldn't need to write everything yourself.
Also, if you plan your own, maybe you could add umami as a temporary solution if possible?

**Reply by JackBailey** [reply]
I'd create my own statistics as Umami wouldn't have enough control over filtering by different clicks on links.

> Also, if you plan your own, maybe you could add umami as a temporary solution if possible?

I could add it so with the variable `FORCE_FRONTEND_REDIRECT` set to true (so the backend isn't handling redirects), it would log an event in umami as "Clicked on link /youtube" for example, would that work for you in the meantime?

**Reply by Hutch79** [reply]
> I'd create my own statistics as Umami wouldn't have enough control over filtering by different clicks on links.
Right, didn't really think this through 😅 

> I could add it so with the variable FORCE_FRONTEND_REDIRECT set to true (so the backend isn't handling redirects), it would log an event in umami as "Clicked on link /youtube" for example, would that work for you in the meantime?

That would be awesome.
This way, it's already possible to filter for specific redirect stats in Umami.

**Reply by JackBailey** [reply]
Okay, [version 1.5.4](https://github.com/Lynx-Shortener/Lynx/commit/0566c3f7f09d07ecee9c6942553b33e62134e793) adds this feature.

You can try it out now using [jackbailey/lynx:1.5.4-beta](https://hub.docker.com/layers/jackbailey/lynx/1.5.4-beta/images/sha256-7d07c4911d77ca823581d541307f1eac89c4dd929f1fe83bd165e9b999678d04?context=explore), let me know if it works fine for you.

**Reply by Hutch79** [reply]
Umami works without any problems!

But I sometimes get stuck on the loading window from Lynx.
If I reload the page, it works.
![grafik](https://github.com/Lynx-Shortener/Lynx/assets/42042811/25fcc062-535f-408e-9953-07b8822e823e)

**Reply by JackBailey** [reply]
I've made a [couple changes](https://github.com/Lynx-Shortener/Lynx/actions/runs/5401480511) to 1.5.4 which should handle errors better.

I've also removed the event for visiting a link as by using FORCE_FRONTEND_REDIRECT the page view for each link is tracked anyway, and you can filter by it.

**Reply by Hutch79** [reply]
> I've also removed the event for visiting [...]
I don't really get what you mean by that ^^'

Better error handling sounds nice!

Is there already a docker image which I could test?

**Reply by JackBailey** [reply]
> > I've also removed the event for visiting [...]

> I don't really get what you mean by that ^^'

Okay so, yesterday I make it so going to a shortened link will show up in the events section in Umami, but as Umami would show this as a page view there is no reason tracking both of them.

> Is there already a docker image which I could test?

Yep, same one - just re-pull it.
